client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/TabLayout.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { FC, useEffect, useState } from 'react';
+import { useIsMediumScreen } from '@common/hooks';
+import {
+  TabContext,
+  TabKeybindings,
+  TabList,
+  ButtonWithIcon,
+} from '@common/components';
+import { PlusCircleIcon } from '@common/icons';
+import {
+  Box,
+  Tab,
+  TableContainer,
+  useTranslation,
+} from '@openmsupply-client/common';
+import { DraftInboundLine } from '../../../../types';
+import { InboundLineEditPanel } from './InboundLineEditPanel';
+import { QuantityTable, PricingTable, LocationTable } from './TabTables';
+
+interface TabLayoutProps {
+  addDraftLine: () => void;
+  draftLines: DraftInboundLine[];
+  isDisabled: boolean;
+  updateDraftLine: (patch: Partial<DraftInboundLine> & { id: string }) => void;
+}
+
+enum Tabs {
+  Batch = 'Batch',
+  Pricing = 'Pricing',
+  Location = 'Location',
+}
+
+export const TabLayout: FC<TabLayoutProps> = ({
+  addDraftLine,
+  draftLines,
+  isDisabled,
+  updateDraftLine,
+}) => {
+  const [currentTab, setCurrentTab] = useState<Tabs>(Tabs.Batch);
+  const [isAdding, setIsAdding] = useState(false);
+  const isMediumScreen = useIsMediumScreen();
+  const t = useTranslation('replenishment');
+
+  const onAdd = () => {
+    setIsAdding(true);
+    addDraftLine();
+  };
+
+  useEffect(() => {
```

Comment:
this is the nub of it - set the flag when adding, and have an effect set it back again when react hits that point in the render cycle